src/main/kotlin/io/zerobase/smarttracing/GraphDao.kt

```diff
@@ -53,4 +57,182 @@ class GraphDao(private val driver: Driver) {
+     * @param long longitude of the site.
+     * @param testing if the site is testing or not.
+     * @param phone contact phone of site manager
+     * @param email contact email of site manager
+     * @param contactName contact name of site manager
+     */
+    fun createSite(id: String, name: String, category: String,
+                   subcategory: String, lat: Float, long: Float,
+                   testing: Boolean, phone: String?, email: String?,
+                   contactName: String?): SiteId? {
+        val query = if (phone == null) {
+            """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: o.phone,
+                        email: o.email,
+                        contactName: o.contactName,
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        } else {
+             """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: '${phone!!}',
+                        email: '${email!!}',
+                        contactName: '${contactName!!}',
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        }
+        return driver.session().use {
+            it.writeTransaction { txn ->
+                val result = txn.run(query).single()["id"].asString()
+                return@writeTransaction result?.let { SiteId(it) }
+            }
+        }
+    }
+
+    /**
+     * Gets all the sites list
+     *
+     * @param id id of the organization
+     */
+    fun getSites(id: String): List<SiteResponse>? {
+        return driver.session().use {
+            it.writeTransaction { txn ->
+                val result = txn.run(
+                        """
+                        MATCH (o:Organization { id: '${id}' })
+                        - [:OWNS] -> (s:Site)
+                        RETURN s.id AS id, s.name AS name
+                        """.trimIndent()
+                ).list { x -> SiteResponse(x["id"].asString(), x["name"].asString()) }
+                return@writeTransaction result?.let { it }
+            }
+        }
+    }
+
+    /**
+     * Creates a scannable for a site.
```

Comment:
Worth maybe explaining that a scannable is an object that a device can connect to, such as a QR code or a BT receiver?

Reply:
Fixed